### PR TITLE
Adding the IO Tab

### PR DESCRIPTION
**Pull request by phixMe**
### Problem

For large graphs users wish to navigate the graph without clicking on the graph itself because it's hard to do with many edges that are not able to be displayed.

One-line summary:
Adds a new tab to move between graph elements without looking at the graph itself.

image

image

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [x] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [x] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [x] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

File: web/src/components/datasets/DatasetDetailPage.tsx [hunk 1]
```diff
@@ -155,14 +156,15 @@ const DatasetDetailPage: FunctionComponent<IProps> = (props) => {
                 {...a11yProps(0)}
                 disableRipple={true}
               />
+              <Tab label={'Inputs/Outputs'} {...a11yProps(1)} disableRipple={true} />
```
Comment (wslulciuc) [inline web/src/components/datasets/DatasetDetailPage.tsx:159]: _Minor_: We can shorten `INPUTS/OUTPUTS` for datasets to `I/O`

File: web/src/components/jobs/JobDetailPage.tsx [hunk 2]
```diff
@@ -98,6 +99,7 @@ const JobDetailPage: FunctionComponent<IProps> = (props) => {
       <Box mb={2} display={'flex'} justifyContent={'space-between'} alignItems={'center'}>
         <Tabs value={tab} onChange={handleChange} textColor='primary' indicatorColor='primary'>
           <Tab label={i18next.t('jobs.latest_tab')} disableRipple={true} />
+          <Tab label={'Inputs/Outputs'} disableRipple={true} />
```
Comment (wslulciuc) [inline web/src/components/jobs/JobDetailPage.tsx:102]: _Minor_: Similarly to datasets, we can shorten `INPUTS/OUTPUTS` for jobs to `I/O`.

**Comment by wslulciuc** [comment]
@phixMe so much great work here to make traversing very large lineage graph easier (or at least, more intuitive). I have some minor feedback on how I see users interacting with the "list" view in a more intuitive way:

1. Under the `I/O` tab, when I click an `IN` or `OUT` edge, I would navigate the user to the `I/O` tab of the selected node
2. When a user selects a node under the `I/O` tab, I'd be great also recenter the graph if the node is not in the current lineage view. I'm not sure on the effort required to do this, so not a must give we do update which edges and nodes are highlight as the user navigates the `I/O` tab.
3. I'm wondering if we should indicate no `IN` or `OUT` edges exist? Currently, it might seem that the `OUT edges are "missing" rather than just not defined.

     Screen Shot 2023-09-22 at 3 17 42 PM

**Comment by phixMe** [comment]
So, we don't have graph recentering available to us, I made an update for the other two items.

**Review by wslulciuc — APPROVED** [review]
LGTM ❤️

**Pull request merged**